syslog-ng-ctl/syslog-ng-ctl.c

```diff
@@ -243,6 +243,31 @@ slng_config(int argc, char *argv[], const gchar *mode, GOptionContext *ctx)
   return res;
 }
 
+static gboolean config_options_start = FALSE;
+static gboolean config_options_stop = FALSE;
+
+static GOptionEntry internal_options[] =
+{
+  { "start", 's', 0, G_OPTION_ARG_NONE, &config_options_start, "start live collection of internal logs", NULL },
+  { "stop", 'x', 0, G_OPTION_ARG_NONE, &config_options_stop, "stop live collection of internal logs", NULL },
+  { NULL,           0,   0, G_OPTION_ARG_NONE, NULL,                         NULL,           NULL }
+};
+
+static gint
+slng_internal(int argc, char *argv[], const gchar *mode, GOptionContext *ctx)
+{
+  GString *cmd = g_string_new("INTERLOGS ");
+
+  if (config_options_start)
+    g_string_append(cmd, "START");
+  else if (config_options_stop)
+    g_string_append(cmd, "STOP");
+
+  gint res = _dispatch_command(cmd->str);
```

Comment:
If we don't specify either start or stop on the CLI, then this code will simply send the "INTERLOGS " string over the socket, which is problematic.

Reply:
@jszigetvari : With the usage of constant strings, this will be resolved.